server/storage/database_test.go

```diff
@@ -501,3 +482,133 @@ func TestDBGetChecksumNotFound(t *testing.T) {
 	require.Error(t, err)
 	require.IsType(t, ErrNotFound{}, err)
 }
+
+func TestGetVersions(t *testing.T) {
+	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+	_, store := SetUpSQLite(t, tempBaseDir)
+	defer os.RemoveAll(tempBaseDir)
+
+	_, repo, _, err := testutils.EmptyRepo("gun")
+	require.NoError(t, err)
+
+	ts, err := repo.SignTimestamp(data.DefaultExpires("timestamp"))
+	tsJSON1, err := json.Marshal(ts)
+	require.NoError(t, err)
+	store.UpdateCurrent(
+		"gun",
+		MetaUpdate{Role: "timestamp", Version: 1, Data: tsJSON1},
+	)
+
+	ts, err = repo.SignTimestamp(data.DefaultExpires("timestamp"))
+	tsJSON2, err := json.Marshal(ts)
+	require.NoError(t, err)
+	store.UpdateCurrent(
+		"gun",
+		MetaUpdate{Role: "timestamp", Version: 2, Data: tsJSON2},
+	)
+	checksumBytes := sha256.Sum256(tsJSON2)
+	checksum := hex.EncodeToString(checksumBytes[:])
+
+	ts, err = repo.SignTimestamp(data.DefaultExpires("timestamp"))
+	tsJSON3, err := json.Marshal(ts)
+	require.NoError(t, err)
+	store.UpdateCurrent(
+		"gun",
+		MetaUpdate{Role: "timestamp", Version: 3, Data: tsJSON3},
+	)
+
+	versions, err := store.GetVersions("gun", "timestamp", "", 0)
+	require.NoError(t, err)
+
+	require.Len(t, versions, 3)
+	require.EqualValues(t, tsJSON3, versions[0])
+	require.EqualValues(t, tsJSON2, versions[1])
+	require.EqualValues(t, tsJSON1, versions[2])
+
+	versions, err = store.GetVersions("gun", "timestamp", checksum, 1)
+	require.NoError(t, err)
+
+	require.Len(t, versions, 1)
+	require.EqualValues(t, tsJSON1, versions[0])
```

Comment:
I totally missed testing what happens if the limit is larger than the total number of possible entries to return. Great catch.